### PR TITLE
stm32f1.c: Export CCM RAM of F303.

### PR DESCRIPTION
**Pull request by UweBonnes**
This allows download to CCM Ram.

### CODE REVIEW

File: src/target/stm32f1.c [hunk 1]
```diff
@@ -142,8 +149,10 @@ bool stm32f1_probe(target *t)
 	case 0x439:  /* STM32F302C8 */
 	case 0x446:  /* STM32F303xD/E and STM32F398xE */
 		t->driver = "STM32F3";
+		target_add_ram(t, 0x10000000, 0x4000);
 		target_add_ram(t, 0x20000000, 0x10000);
-		stm32f1_add_flash(t, 0x8000000, 0x80000, 0x800);
+		flash_size= (target_mem_read32(t, FLASHSIZE) & 0xffff) * 0x400;
```
Comment (gsmcmullin) [inline src/target/stm32f1.c]: Why this change?
- It's not declared in the PR description or the commit message.
- It appears to be inconsistent with your newly added comment above, that also has nothing to do with CCM RAM.
- Put spaces on both sides of `=`.
- `FLASHSIZE` is 0x1FFFF7E0, but the register is at 0x1FFFF7CC, is it not?  Did you test this?

**Comment by UweBonnes** [comment]
I wanted to use a flash like linker script to upload and run totally in RAM. The setup should get things right after a reset, so initialized data must be loaded to one region by the debugger and copied  by thecontroller on startup. CCM ram as a different memory region is needed for that task. It was not exported before, so download was refused by the debugger

When doing so, I revisited to probe(), but after some fiddling I saw that idcodes and devices are mixed up like herb and beet. I forgot to revoke these changes, perhaps with the hope that new new setup was "clearer".

In the spirit of " one patch for one task" the patch now only adds CCM ram.

B.t.w, do you think that "run from RAM, resistant to reset and without code change" is possible with only one RAM region?

**Comment by gsmcmullin** [comment]
I don't see why you would need more than one ram region.  You will just need enough ram that you can afford to have 2 copies of your `.data` section, they can be located in the same physical region of memory.

**Comment by UweBonnes** [comment]
As I understand the gnu linker, having two copies of a section is not possible.
